src/libraries/System.IO.Pipes/src/System/IO/Pipes/PipeStream.Windows.cs

```diff
@@ -16,6 +16,218 @@ public abstract partial class PipeStream : Stream
         internal const bool CheckOperationsRequiresSetHandle = true;
         internal ThreadPoolBoundHandle? _threadPoolBinding;
 
+        public override int Read(byte[] buffer, int offset, int count)
+        {
+            if (_isAsync)
+            {
+                return ReadAsync(buffer, offset, count, CancellationToken.None).GetAwaiter().GetResult();
```

Comment:
I know this was just copied from before; but why is this `ReadAsync(...).GetAwaiter().GetResult()` and `Read(Span)` is `base.Read(buffer)`?

Couldn't this also be 
```csharp
return base.Read(new Span(buffer, offset, count));
```
Or is the other one wrong?

Reply:
@benaadams the `Stream.Read(Span)` method has an additional overhead of renting an array from the `ArrayPool` and copying the memory:

https://github.com/dotnet/runtime/blob/0f64b267ac0552f07b06f18103581b880c8c53c6/src/libraries/System.Private.CoreLib/src/System/IO/Stream.cs#L659-L677

this is why we use `Read(Array)` here and in few other places like `FileStream`

Reply:
Ah, and then it calls into the above overload using the array